### PR TITLE
Refactor navbar

### PR DESCRIPTION
**Pull request by chownces**
### Description

- Fix CSS issues from react-router upgrade
- Improve readability of navbar component

### Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Code quality improvements

### Checklist

- [x] I have tested this code

### CODE REVIEW

File: src/commons/navigationBar/subcomponents/AcademyNavigationBar.tsx [hunk 1]
```diff
@@ -152,4 +51,86 @@ export const icons: IconName[] = [
...
+      disabled: !isAdmin,
+      hiddenInBreakpoints: ['xs', 'sm', 'md', 'lg']
+    },
+    {
+      to: `/courses/${courseId}/adminpanel`,
+      icon: IconNames.SETTINGS,
+      text: 'Admin Panel',
+      disabled: !isAdmin,
+      hiddenInBreakpoints: ['xs', 'sm', 'md', 'lg']
+    }
```
Comment (RichDom2185) [inline src/commons/navigationBar/subcomponents/AcademyNavigationBar.tsx:132]: Are the ordering changes (originally they were XP, Grading, Story, Admin Panel) intentionally part of the UX improvements?

File: src/commons/navigationBar/subcomponents/AcademyNavigationBar.tsx [hunk 2]
```diff
@@ -152,4 +51,86 @@ export const icons: IconName[] = [
...
+      text: 'Story Simulator',
+      disabled: !isStaffOrAdmin,
+      hiddenInBreakpoints: ['xs', 'sm', 'md']
+    },
+    {
+      to: `/courses/${courseId}/xpcalculation`,
+      icon: IconNames.CALCULATOR,
+      text: 'XP Calculation',
+      disabled: !isAdmin,
+      hiddenInBreakpoints: ['xs', 'sm', 'md', 'lg']
```
Comment (RichDom2185) [inline src/commons/navigationBar/subcomponents/AcademyNavigationBar.tsx:124]: Likewise, originally, XP Calculation was only hidden in the `xs` and `sm` breakpoints. Is this change intentional as part of the improvements to readability of NavBar? I'm not sure if this refers to code readability of UX improvements. It definitely does improve the code readability, but in terms of the UX, was this an intentional change? Just making sure haha

File: src/commons/navigationBar/subcomponents/AcademyNavigationBar.tsx [hunk 3]
```diff
@@ -152,4 +51,86 @@ export const icons: IconName[] = [
...
+export const assessmentTypesToNavlinkInfo = ({
+  assessmentTypes,
+  courseId,
+  isEnrolledInACourse
+}: {
+  assessmentTypes?: string[];
+  courseId?: number;
+  isEnrolledInACourse: boolean;
+}): NavbarEntryInfo[] =>
+  (assessmentTypes || []).map((assessmentType, idx) => ({
```
Comment (RichDom2185) [inline src/commons/navigationBar/subcomponents/AcademyNavigationBar.tsx:63]:
```suggestion
  assessmentTypes = [],
  courseId,
  isEnrolledInACourse
}: {
  assessmentTypes?: string[];
  courseId?: number;
  isEnrolledInACourse: boolean;
}): NavbarEntryInfo[] =>
  assessmentTypes.map((assessmentType, idx) => ({
```

File: src/commons/navigationBar/subcomponents/AcademyNavigationBar.tsx [hunk 4]
```diff
@@ -152,4 +51,86 @@ export const icons: IconName[] = [
...
+    to: `/courses/${courseId}/${assessmentTypeLink(assessmentType)}`,
```
Comment (RichDom2185) [inline src/commons/navigationBar/subcomponents/AcademyNavigationBar.tsx:64]: What happens if `courseId` is `undefined`?

File: src/commons/navigationBar/NavigationBar.tsx [hunk 5]
```diff
@@ -65,277 +55,222 @@ const NavigationBar: React.FC = () => {
...
+        )}
+        <NavLink
+          className={classNames('NavigationBar__link', Classes.BUTTON, Classes.MINIMAL)}
+          to={
+            Constants.playgroundOnly ? '/' : courseId == null ? '/welcome' : `/courses/${courseId}`
+          }
+        >
+          <Icon icon={IconNames.SYMBOL_DIAMOND} />
+          <NavbarHeading style={{ paddingBottom: '0px' }}>
+            {courseShortName || 'Source Academy'}
```
Comment (RichDom2185) [inline src/commons/navigationBar/NavigationBar.tsx:122]: Can we define a default fallback value in `Constants.ts` instead of hardcoding it here? It would make it easier for not only customisation, but also i18n efforts.

File: src/commons/navigationBar/NavigationBar.tsx [hunk 6]
```diff
@@ -65,277 +55,222 @@ const NavigationBar: React.FC = () => {
...
-          {desktopLogoButton}
+          <NavLink
+            className={classNames('NavigationBar__link', Classes.BUTTON, Classes.MINIMAL, {
+              [Classes.ACTIVE]: highlightDesktopLogo(location)
+            })}
+            to={courseId == null ? '/welcome' : `/courses/${courseId}`}
+          >
+            <Icon icon={IconNames.SYMBOL_DIAMOND} />
+            <NavbarHeading style={{ paddingBottom: '0px' }}>
+              {courseShortName || 'Source Academy'}
```
Comment (RichDom2185) [inline src/commons/navigationBar/NavigationBar.tsx:261]: Ditto. Noticed that "Source Academy @ NUS" is also no more. Perhaps the default fallback value can be done conditionally?

**Review by RichDom2185 — COMMENTED** [review]
Looks good! The codebase is so much more readable now 😄 I only have some minor comments as follows.

Thanks for this long-overdue refactor!

File: src/commons/navigationBar/subcomponents/AcademyNavigationBar.tsx (hunk 1 above)
Comment (chownces) [reply src/commons/navigationBar/subcomponents/AcademyNavigationBar.tsx:132]: yes. lets keep the admin related routes to the right

File: src/commons/navigationBar/NavigationBar.tsx (hunk 6 above)
Comment (chownces) [reply src/commons/navigationBar/NavigationBar.tsx:261]: This is done on purpose to take out the hardcoded @ NUS. Replaced with a more general purposed 'Source Academy'

File: src/commons/navigationBar/subcomponents/AcademyNavigationBar.tsx (hunk 4 above)
Comment (chownces) [reply src/commons/navigationBar/subcomponents/AcademyNavigationBar.tsx:64]: This component is only rendered when `role` is defined, in which case `courseId` should also be defined. To make it more robust, I added an additional early return to not render this component when `courseId` or `role` is undefined

File: src/commons/navigationBar/NavigationBar.tsx (hunk 6 above)
Comment (RichDom2185) [reply src/commons/navigationBar/NavigationBar.tsx:261]: Noted, that makes sense. I'm just thinking of potential integrations with i18n, hence my suggestion on putting this into `Constants.tsx` or something like that. But on second though, let us not prematurely do all these refactoring. Thanks!

**Review by RichDom2185 — APPROVED** [review]
LGTM, thanks!
